src/components/organisms/Results.vue

```diff
@@ -107,27 +128,43 @@ export default defineComponent({
         removeFunc = clearMetaDataDrawer;
       }
 
-      if (!searchResults.value) {
-        return null;
-      }
       if (value && addFunc) {
-        addFunc(spatialId);
+        addFunc(spatialId, props.regions);
       } else if (removeFunc) {
-        removeFunc(spatialId);
+        removeFunc(spatialId, props.regions);
       }
-      searchResults.value = searchResults.value.map((entry) => {
-        if (entry.spatial_id === spatialId) {
-          // eslint-disable-next-line @typescript-eslint/camelcase
-          return Object.assign(entry, { [fieldName]: value });
-        }
-        return entry;
-      });
+
+      if (!props.regions) {
+        if (!searchResults.value) return null;
+        searchResults.value = searchResults.value.map(
+          (entry) => {
+            if (entry.spatial_id === spatialId) {
+              return Object.assign(entry, { [fieldName]: value });
```

Comment:
Rather than do an in-place assignment inside a `map` body, let's stick with functional paradigms and do

```suggestion
              return {...entry, [fieldName]: value };
```